ambry-router/src/test/java/com.github.ambry.router/RouterTestHelpers.java

```diff
@@ -33,5 +42,138 @@ static boolean haveEquivalentFields(BlobProperties a, BlobProperties b) {
+   *                                servers are set to {@link ServerErrorCode#No_Error}
+   * @param serverLayout A {@link MockServerLayout} that is used to find the {@link MockServer}s to set error codes on.
+   * @param expectedError The {@link RouterErrorCode} expected, or null if no error is expected.
+   * @param errorCodeChecker Performs the checks that ensure that the expected {@link RouterErrorCode} is returned .
+   * @throws Exception
+   */
+  static void testWithErrorCodes(ServerErrorCode[] serverErrorCodesInOrder, PartitionId partition,
+      MockServerLayout serverLayout, RouterErrorCode expectedError, ErrorCodeChecker errorCodeChecker)
+      throws Exception {
+    setServerErrorCodes(serverErrorCodesInOrder, partition, serverLayout);
+    errorCodeChecker.check(expectedError);
+    resetServerErrorCodes(serverLayout);
+  }
+
+  /**
+   * Set the servers in the specified layout to respond with the designated error codes.
+   * @param serverErrorCodesInOrder The error codes to set in the order of the servers in {@code serverLayout}.  If
+   *                                there are more values in this array than there are servers, an exception is thrown.
+   *                                If there are less, the rest of the servers are set to
+   *                                {@link ServerErrorCode#No_Error}
+   * @param serverLayout A {@link MockServerLayout} that is used to find the {@link MockServer}s to set error codes on.
+   * @throws Exception
+   */
+  static void setServerErrorCodes(ServerErrorCode[] serverErrorCodesInOrder, MockServerLayout serverLayout) {
+    Collection<MockServer> servers = serverLayout.getMockServers();
+    if (serverErrorCodesInOrder.length > servers.size()) {
+      throw new IllegalArgumentException("More server error codes provided than servers in cluster");
+    }
+    int i = 0;
+    for (MockServer server : servers) {
+      if (i < serverErrorCodesInOrder.length) {
+        server.setServerErrorForAllRequests(serverErrorCodesInOrder[i++]);
+      } else {
+        server.setServerErrorForAllRequests(ServerErrorCode.No_Error);
+      }
+    }
+  }
+
+  /**
+   * Set the servers in the specified partition to respond with the designated error codes.
+   * @param serverErrorCodesInOrder The error codes to set in the order of the servers returned by
+   *                                {@link PartitionId#getReplicaIds()}. If there are more values in this array than
+   *                                there are servers, an exception is thrown. If there are less, the rest of the
+   *                                servers are set to {@link ServerErrorCode#No_Error}
+   * @param serverLayout A {@link MockServerLayout} that is used to find the {@link MockServer}s to set error codes on.
+   * @throws Exception
+   */
+  static void setServerErrorCodes(ServerErrorCode[] serverErrorCodesInOrder, PartitionId partition,
```

Comment:
Some of the DeleteManager tests rely on the error codes being set in the order of the replicas returned by a partition.